packages/yew-router/src/macro_helpers.rs

```diff
@@ -9,19 +9,28 @@ pub fn build_router<R: Routable>() -> Router {
     let base = base_url();
     let mut router = Router::new();
     R::routes().iter().for_each(|path| {
-        let path = match &base {
-            Some(base) if base != "/" => build_path_with_base(path),
-            _ => path.to_string(),
+        match &base {
+            Some(base) if base != "/" => {
+                let route = build_path_with_base(path);
+                let mut dest = route.replace(base, "");
```

Comment:
This seems very convoluted... Are you sure you can't simplify the entire thing a bit?
Particularly I would love to see a solution that's more easily verifiable to be correct.

Reply:
I can't think of a better solution. The only other solution that comes to my mind is using base in the generated code but that is too much unnecessary overhead. The base isn't a literal so the left side of match statement can't be computed at compile time.  
If you know of a better solution, I'd love to hear it.